plugins/vsphere/cloudkeeper_plugin_vsphere/resources.py

```diff
@@ -0,0 +1,110 @@
+from cloudkeeper.graph import Graph
+import cloudkeeper.logging
+from cloudkeeper.baseresources import (
+    BaseAccount,
+    BaseRegion,
+    BaseResource,
+    BaseInstance,
+    InstanceStatus,
+)
+
+from pyVmomi import vim
+
+from .vsphere_client import new_vsphere_client
+
+log = cloudkeeper.logging.getLogger("cloudkeeper." + __name__)
+
+
+class VSphereCluster(BaseAccount):
+    resource_type = "vsphere_cluster"
+
+    def delete(self, graph: Graph) -> bool:
+        return NotImplemented
+
+
+class VSphereDataCenter(BaseRegion):
+    resource_type = "vsphere_data_center"
+
+    def delete(self, graph: Graph) -> bool:
+        return NotImplemented
+
+
+class VSphereResource(BaseResource):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    resource_type = "vsphere_resource"
+
+    @property
+    def vsphere_client(self):
+        return new_vsphere_client()
+
+    def delete(self, graph: Graph) -> bool:
+        log.debug(
+            f"Deleting resource {self.id} in account {self.account(graph).id} region {self.region(graph).id}"
+        )
+        return True
+
+    def update_tag(self, key, value) -> bool:
+        log.debug(f"Updating or setting tag {key}: {value} on resource {self.id}")
+        return True
+
+    def delete_tag(self, key) -> bool:
+        log.debug(f"Deleting tag {key} on resource {self.id}")
+        return True
+
+
+class VSphereInstance(BaseInstance, VSphereResource):
+    resource_type = "vsphere_instance"
+
+    instance_status_map = {
```

Comment:
Is this status_map now accurate for Vsphere or are some of the states remnants from the example collector?